### ISSUE TITLE
Update ESLint

### ISSUE THREAD

**Issue by floatplane** [NWACus/avy#359]
running `yarn ci:eslint` produces:
```
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.0.0

YOUR TYPESCRIPT VERSION: 5.1.3

Please only submit bug reports when using the officially supported version.

=============
```

**Comment by jaas666** [comment]
Related: https://github.com/typescript-eslint/typescript-eslint/issues/6934

**Comment by yuliadub** [comment]
annoyingly enough when I run the command on windows I dont get a warning :/

![image](https://github.com/NWACus/avy/assets/9418602/ac383436-6031-4046-bade-53d53497a817)

**Comment by floatplane** [comment]
oof, thanks for that link @jaas666. that's kind of scary.

@yuliadub I think this highlights that our script needs fixing - the call to `eslint` is picking up the global ESLint and not whatever version is specified locally in `package.json`

https://github.com/NWACus/avy/blob/f72d8ff3730658154518bc6d130789143ef2bb36/scripts/eslint.sh#L3-L4

Note that I get a different version depending on whether I run `yarn eslint` or `eslint`:

```sh
~/projects/avalanche-forecast preview *20 ········································································································································   system
❯ yarn eslint --version
yarn run v1.22.19
$ /Users/brians/projects/avalanche-forecast/node_modules/.bin/eslint --version
v8.49.0
✨  Done in 0.72s.

~/projects/avalanche-forecast preview *20 ········································································································································   system
❯  eslint --version    
v8.48.0
```

I'm curious which version you have installed globally?

I will fix this now.

**Comment by floatplane** [comment]
Nothing more to do until `typescript-eslint` is revved

**Comment by stevekuznetsov** [comment]
The package seems to have new releases published [almost daily](https://github.com/typescript-eslint/typescript-eslint/releases) so it would be good to check if we can update this now.

**Comment by stevekuznetsov** [comment]
This was fixed in #674
